CMakeLists.txt

```diff
@@ -376,26 +403,73 @@ foreach (s ${COMPILETIME_OPTIONS_LIST})
 	set (COMPILETIME_OPTIONS "${COMPILETIME_OPTIONS} -D${s}")
 endforeach (s)
 
-configure_file(libtorrent-rasterbar-cmake.pc.in libtorrent-rasterbar.pc)
+generate_and_install_pkg_config_file(torrent-rasterbar)
 
-string (COMPARE EQUAL "${CMAKE_SIZEOF_VOID_P}" "8" IS64BITS)
+install(TARGETS torrent-rasterbar EXPORT LibtorrentRasterbarTargets
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
+install(DIRECTORY include/libtorrent DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} FILES_MATCHING PATTERN "*.h*")
+
+# === generate a CMake Config File ===
+include(CMakePackageConfigHelpers)
+set(ConfigPackageLocation ${CMAKE_INSTALL_LIBDIR}/cmake/LibtorrentRasterbar)
+string(REGEX REPLACE "([^;]+)" "find_dependency(\\1)" _find_dependency_calls "${_package_dependencies}")
+string(REPLACE ";" "\n" _find_dependency_calls "${_find_dependency_calls}")
+
+write_basic_package_version_file(
+	"${CMAKE_CURRENT_BINARY_DIR}/LibtorrentRasterbar/LibtorrentRasterbarConfigVersion.cmake"
+	VERSION ${libtorrent_VERSION}
+	COMPATIBILITY SameMinorVersion
+)
 
-if (IS64BITS AND RESPECTLIB64)
-	set (LIBDIR "lib64")
-else()
-	set (LIBDIR "lib")
-endif()
+export(EXPORT LibtorrentRasterbarTargets
+	FILE "${CMAKE_CURRENT_BINARY_DIR}/LibtorrentRasterbar/LibtorrentRasterbarTargets.cmake"
+	NAMESPACE LibtorrentRasterbar::
+)
 
-install(TARGETS torrent-rasterbar DESTINATION ${LIBDIR})
-install(DIRECTORY include/libtorrent
-	DESTINATION include
-	PATTERN ".svn" EXCLUDE)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libtorrent-rasterbar.pc DESTINATION ${LIBDIR}/pkgconfig)
+configure_package_config_file(LibtorrentRasterbarConfig.cmake.in
```

Comment:
when I try this locally (on macOS), cmake cannot find this file:

```
LibtorrentRasterbarConfig.cmake.in does not exist
```

did you forget to add it?

Reply:
it seems to be looking for it in the root of libtorrent

Reply:
the full error is:
```
CMake Error: File /Users/arvid/Documents/dev/2/libtorrent/LibtorrentRasterbarConfig.cmake.in does not exist.
CMake Error at /usr/local/Cellar/cmake/3.12.3/share/cmake/Modules/CMakePackageConfigHelpers.cmake:316 (configure_file):
  configure_file Problem configuring file
Call Stack (most recent call first):
  CMakeLists.txt:432 (configure_package_config_file)
```

Reply:
Sorry, did not notice that this file type is covered by .gitignore. Fixed.